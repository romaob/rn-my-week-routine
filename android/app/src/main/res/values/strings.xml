<resources>
    <string name="app_name">MyWeekRoutine</string>
</resources>
