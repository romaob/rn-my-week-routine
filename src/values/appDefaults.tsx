export enum APP_KEYS {
  LANGUAGE = '@myweekroutine.language',
  ALERTS_ENABLED = '@myweekroutine.alertsEnabled',
  ONBOARDING = '@myweekroutine.onboarding',
  EVENTS = '@myweekroutine.events',
}

export interface Event {
  id: string;
  name: string;
  indexes: number[];
  description: string;
  startAt: string;
  endAt: string;
  alertEnabled: boolean;
  alertSent: boolean;
  alertConfirmed: boolean;
  added: string;
  updated: string;
}
