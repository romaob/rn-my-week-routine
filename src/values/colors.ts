// App color palette
export const colors = {
  light: {
    primary: '#3F51B5',
    secondary: '#536DFE',
    accent: '#FF4081',
    danger: '#f44336',
    warning: '#ff9800',
    success: '#4caf50',
    info: '#2196f3',
    background: '#ffffff',
    cardBackground: '#ffffff',
    disabled: '#bdbdbd',
    text: '#000000',
    textContrast: '#ffffff',
    textSecondary: '#757575',
    textDisabled: '#bdbdbd',
  },
};
