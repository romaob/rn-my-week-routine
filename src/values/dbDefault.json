{
    "language": "en",
    "alertsEnabled": false,
    "onboarding": false,
    "events": []
}