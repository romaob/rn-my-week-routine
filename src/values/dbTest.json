{
    "language": "en",
    "alertsEnabled": true,
    "onboarding": false,
    "events": [
        {
            "id": 1,
            "name": "Event 1",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 1 description",
            "startAt": "00:00:00",
            "endAt": "01:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 2,
            "name": "Event 2",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 2 description",
            "startAt": "01:00:00",
            "endAt": "02:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 3,
            "name": "Event 3",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 3 description",
            "startAt": "02:00:00",
            "endAt": "03:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 4,
            "name": "Event 4",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 4 description",
            "startAt": "03:00:00",
            "endAt": "04:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        },
        {
            "id": 5,
            "name": "Event 5",
            "indexes": [0,1,2,3,4,5,6],
            "description": "Event 5 description",
            "startAt": "04:00:00",
            "endAt": "05:00:00",
            "alertEnabled": true,
            "alertSent": false,
            "alertConfirmed": false,
            "added": "2015-01-01T00:00:00.000Z",
            "updated": "2015-01-01T00:00:00.000Z"
        }
    ]
}