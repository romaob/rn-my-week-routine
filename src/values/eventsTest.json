[{"id":"0test1","name":"Event A0","indexes":[0,1,2,3,4,5],"description":"Event A0 description","startAt":"2023-09-06T03:00:55.695Z","endAt":"2023-09-06T07:00:55.695Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.695Z","updated":"2023-09-06T16:33:55.695Z"},{"id":"0test2","name":"Event B0","indexes":[0,1,2,3,4,6],"description":"Event B0 description","startAt":"2023-09-06T04:00:55.695Z","endAt":"2023-09-06T08:00:55.695Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.695Z","updated":"2023-09-06T16:33:55.695Z"},{"id":"0test3","name":"Event C0","indexes":[0,1,2,3,5,6],"description":"Event C0 description","startAt":"2023-09-06T05:00:55.695Z","endAt":"2023-09-06T09:00:55.696Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.696Z","updated":"2023-09-06T16:33:55.696Z"},{"id":"0test4","name":"Event D0","indexes":[0,1,2,4,5,6],"description":"Event D0 description","startAt":"2023-09-06T06:00:55.696Z","endAt":"2023-09-06T10:00:55.696Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.696Z","updated":"2023-09-06T16:33:55.696Z"},{"id":"0test5","name":"Event E0","indexes":[0,1,3,4,5,6],"description":"Event E0 description","startAt":"2023-09-06T05:00:55.696Z","endAt":"2023-09-06T06:00:55.696Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.696Z","updated":"2023-09-06T16:33:55.696Z"},{"id":"1test1","name":"Event A1","indexes":[0,1,2,3,4,5],"description":"Event A1 description","startAt":"2023-09-06T07:00:55.696Z","endAt":"2023-09-06T11:00:55.696Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.697Z","updated":"2023-09-06T16:33:55.697Z"},{"id":"1test2","name":"Event B1","indexes":[0,1,2,3,4,6],"description":"Event B1 description","startAt":"2023-09-06T08:00:55.697Z","endAt":"2023-09-06T12:00:55.697Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.697Z","updated":"2023-09-06T16:33:55.697Z"},{"id":"1test3","name":"Event C1","indexes":[0,1,2,3,5,6],"description":"Event C1 description","startAt":"2023-09-06T09:00:55.697Z","endAt":"2023-09-06T13:00:55.697Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.697Z","updated":"2023-09-06T16:33:55.697Z"},{"id":"1test4","name":"Event D1","indexes":[0,1,2,4,5,6],"description":"Event D1 description","startAt":"2023-09-06T10:00:55.697Z","endAt":"2023-09-06T14:00:55.697Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.698Z","updated":"2023-09-06T16:33:55.698Z"},{"id":"1test5","name":"Event E1","indexes":[0,1,3,4,5,6],"description":"Event E1 description","startAt":"2023-09-06T09:00:55.698Z","endAt":"2023-09-06T10:00:55.698Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.698Z","updated":"2023-09-06T16:33:55.698Z"},{"id":"2test1","name":"Event A2","indexes":[0,1,2,3,4,5],"description":"Event A2 description","startAt":"2023-09-06T11:00:55.698Z","endAt":"2023-09-06T15:00:55.698Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.698Z","updated":"2023-09-06T16:33:55.698Z"},{"id":"2test2","name":"Event B2","indexes":[0,1,2,3,4,6],"description":"Event B2 description","startAt":"2023-09-06T12:00:55.698Z","endAt":"2023-09-06T16:00:55.698Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.698Z","updated":"2023-09-06T16:33:55.698Z"},{"id":"2test3","name":"Event C2","indexes":[0,1,2,3,5,6],"description":"Event C2 description","startAt":"2023-09-06T13:00:55.698Z","endAt":"2023-09-06T17:00:55.699Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.699Z","updated":"2023-09-06T16:33:55.699Z"},{"id":"2test4","name":"Event D2","indexes":[0,1,2,4,5,6],"description":"Event D2 description","startAt":"2023-09-06T14:00:55.699Z","endAt":"2023-09-06T18:00:55.699Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.699Z","updated":"2023-09-06T16:33:55.699Z"},{"id":"2test5","name":"Event E2","indexes":[0,1,3,4,5,6],"description":"Event E2 description","startAt":"2023-09-06T13:00:55.699Z","endAt":"2023-09-06T14:00:55.699Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.699Z","updated":"2023-09-06T16:33:55.699Z"},{"id":"3test1","name":"Event A3","indexes":[0,1,2,3,4,5],"description":"Event A3 description","startAt":"2023-09-06T15:00:55.699Z","endAt":"2023-09-06T19:00:55.699Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.700Z","updated":"2023-09-06T16:33:55.700Z"},{"id":"3test2","name":"Event B3","indexes":[0,1,2,3,4,6],"description":"Event B3 description","startAt":"2023-09-06T16:00:55.700Z","endAt":"2023-09-06T20:00:55.700Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.700Z","updated":"2023-09-06T16:33:55.700Z"},{"id":"3test3","name":"Event C3","indexes":[0,1,2,3,5,6],"description":"Event C3 description","startAt":"2023-09-06T17:00:55.700Z","endAt":"2023-09-06T21:00:55.700Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.700Z","updated":"2023-09-06T16:33:55.700Z"},{"id":"3test4","name":"Event D3","indexes":[0,1,2,4,5,6],"description":"Event D3 description","startAt":"2023-09-06T18:00:55.700Z","endAt":"2023-09-06T22:00:55.700Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.700Z","updated":"2023-09-06T16:33:55.700Z"},{"id":"3test5","name":"Event E3","indexes":[0,1,3,4,5,6],"description":"Event E3 description","startAt":"2023-09-06T17:00:55.700Z","endAt":"2023-09-06T18:00:55.701Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.701Z","updated":"2023-09-06T16:33:55.701Z"},{"id":"4test1","name":"Event A4","indexes":[0,1,2,3,4,5],"description":"Event A4 description","startAt":"2023-09-06T19:00:55.701Z","endAt":"2023-09-06T23:00:55.701Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.701Z","updated":"2023-09-06T16:33:55.701Z"},{"id":"4test2","name":"Event B4","indexes":[0,1,2,3,4,6],"description":"Event B4 description","startAt":"2023-09-06T20:00:55.701Z","endAt":"2023-09-07T00:00:55.701Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.701Z","updated":"2023-09-06T16:33:55.701Z"},{"id":"4test3","name":"Event C4","indexes":[0,1,2,3,5,6],"description":"Event C4 description","startAt":"2023-09-06T21:00:55.701Z","endAt":"2023-09-07T01:00:55.701Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.702Z","updated":"2023-09-06T16:33:55.702Z"},{"id":"4test4","name":"Event D4","indexes":[0,1,2,4,5,6],"description":"Event D4 description","startAt":"2023-09-06T22:00:55.702Z","endAt":"2023-09-07T02:00:55.702Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.702Z","updated":"2023-09-06T16:33:55.702Z"},{"id":"4test5","name":"Event E4","indexes":[0,1,3,4,5,6],"description":"Event E4 description","startAt":"2023-09-06T21:00:55.702Z","endAt":"2023-09-06T22:00:55.702Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.702Z","updated":"2023-09-06T16:33:55.702Z"},{"id":"5test1","name":"Event A5","indexes":[0,1,2,3,4,5],"description":"Event A5 description","startAt":"2023-09-06T23:00:55.702Z","endAt":"2023-09-07T03:00:55.702Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.702Z","updated":"2023-09-06T16:33:55.702Z"},{"id":"5test2","name":"Event B5","indexes":[0,1,2,3,4,6],"description":"Event B5 description","startAt":"2023-09-07T00:00:55.702Z","endAt":"2023-09-07T03:00:55.703Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.703Z","updated":"2023-09-06T16:33:55.703Z"},{"id":"5test3","name":"Event C5","indexes":[0,1,2,3,5,6],"description":"Event C5 description","startAt":"2023-09-07T01:00:55.703Z","endAt":"2023-09-07T03:00:55.703Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.703Z","updated":"2023-09-06T16:33:55.703Z"},{"id":"5test4","name":"Event D5","indexes":[0,1,2,4,5,6],"description":"Event D5 description","startAt":"2023-09-07T02:00:55.703Z","endAt":"2023-09-07T03:00:55.703Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.704Z","updated":"2023-09-06T16:33:55.704Z"},{"id":"5test5","name":"Event E5","indexes":[0,1,3,4,5,6],"description":"Event E5 description","startAt":"2023-09-07T01:00:55.704Z","endAt":"2023-09-07T02:00:55.704Z","alertEnabled":false,"alertSent":false,"alertConfirmed":false,"added":"2023-09-06T16:33:55.704Z","updated":"2023-09-06T16:33:55.704Z"}]