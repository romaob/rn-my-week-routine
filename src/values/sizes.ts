//Options for font sizes, padding, margin, etc.
export const sizes = {
  // Font sizes
  font: {
    sm: 12,
    md: 14,
    md2: 18,
    lg: 24,
    xl: 32,
  },
  // Padding sizes
  padding: {
    sm: 8,
    md: 16,
    lg: 24,
  },
  // Margin sizes
  margin: {
    sm: 8,
    md: 16,
    lg: 24,
  },
  // App values
  appValues: {
    timeSlotHeight: 30,
  },
};
