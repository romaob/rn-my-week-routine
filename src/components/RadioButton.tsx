import React from 'react';
import {View, StyleSheet} from 'react-native';
import Label from './Label';
import {sizes} from '../values/sizes';
import {TouchableOpacity} from 'react-native-gesture-handler';
import {colors} from '../values/colors';

export interface RadioButtonProps {
  label?: string;
  size?: number;
  selected: boolean;
  onPress: () => void;
}

export default function RadioButton({
  label,
  size = sizes.font.md,
  selected,
  onPress,
}: RadioButtonProps): JSX.Element {
  return (
    <TouchableOpacity
      testID="radioButton"
      style={styles.container}
      onPress={onPress}>
      <View
        testID="radioButtonCircle"
        style={{
          ...styles.radio,
          ...{padding: size / 4},
        }}>
        <View
          testID="radioButtonCircleInner"
          style={{
            ...styles.radioInner,
            ...(selected ? styles.radioSelected : {}),
            ...(size
              ? {width: size, height: size, borderRadius: size / 2}
              : {}),
          }}
        />
      </View>
      <Label
        testID="radioButtonLabel"
        text={label || ''}
        style={{
          ...styles.label,
          ...{fontSize: size},
        }}
      />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: sizes.padding.sm,
  },
  radio: {
    display: 'flex',
    borderWidth: 2,
    borderColor: colors.light.accent,
    backgroundColor: 'white',
    borderRadius: 999,
  },
  radioInner: {
    backgroundColor: 'transparent',
  },
  radioSelected: {
    backgroundColor: colors.light.accent,
    borderRadius: 999,
  },
  label: {},
});
