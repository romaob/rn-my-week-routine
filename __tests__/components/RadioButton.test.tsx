import React from 'react';

import {fireEvent, render} from '@testing-library/react-native';
import {colors} from '../../src/values/colors';
import RadioButton from '../../src/components/RadioButton';

describe('RadioButton render tests', () => {
  it('should render the radio button', () => {
    const result = render(<RadioButton selected={false} onPress={() => {}} />);
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircle').props.style.borderWidth,
    ).not.toBe(0);
    expect(
      result.getByTestId('radioButtonCircle').props.style.borderColor,
    ).toBe(colors.light.accent);
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe('transparent');
  });

  it('should render the radio button selected', () => {
    const result = render(<RadioButton selected={true} onPress={() => {}} />);
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircle').props.style.borderWidth,
    ).not.toBe(0);
    expect(
      result.getByTestId('radioButtonCircle').props.style.borderColor,
    ).toBe(colors.light.accent);
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe(colors.light.accent);
  });

  it('should render the radio button with label', () => {
    const text = 'Test';
    const result = render(
      <RadioButton label={text} selected={false} onPress={() => {}} />,
    );
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(result.getByTestId('radioButtonLabel')).toBeDefined();
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe('transparent');
    expect(result.getByText(text)).toBeDefined();
    expect(
      result
        .getByTestId('radioButtonLabel')
        .props.children.some((child: any) => child === text),
    ).toBeTruthy();
  });

  it('should render the radio button with label and selected', () => {
    const text = 'Test';
    const result = render(
      <RadioButton label={text} selected={true} onPress={() => {}} />,
    );
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(result.getByTestId('radioButtonLabel')).toBeDefined();
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe(colors.light.accent);
    expect(result.getByText(text)).toBeDefined();
    expect(
      result
        .getByTestId('radioButtonLabel')
        .props.children.some((child: any) => child === text),
    ).toBeTruthy();
  });

  it('should render the radio button with label and size', () => {
    const text = 'Test';
    const size = 20;
    const result = render(
      <RadioButton
        label={text}
        size={size}
        selected={false}
        onPress={() => {}}
      />,
    );
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(result.getByTestId('radioButtonLabel')).toBeDefined();
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe('transparent');
    expect(result.getByText(text)).toBeDefined();
    expect(
      result
        .getByTestId('radioButtonLabel')
        .props.children.some((child: any) => child === text),
    ).toBeTruthy();
    expect(result.getByTestId('radioButtonCircle').props.style.padding).toBe(
      size / 4,
    );
    expect(result.getByTestId('radioButtonCircleInner').props.style.width).toBe(
      size,
    );
  });
});

describe('RadioButton action tests', () => {
  it('should call onPress when radio button is pressed', () => {
    const mockOnPress = jest.fn();
    const result = render(
      <RadioButton selected={false} onPress={mockOnPress} />,
    );
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe('transparent');
    expect(mockOnPress).toHaveBeenCalledTimes(0);
    fireEvent.press(result.getByTestId('radioButton'));
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('should call onPress when radio button is pressed and selected', () => {
    const mockOnPress = jest.fn();
    const result = render(
      <RadioButton selected={true} onPress={mockOnPress} />,
    );
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
    expect(result.getByTestId('radioButton')).toBeDefined();
    expect(result.getByTestId('radioButtonCircle')).toBeDefined();
    expect(result.getByTestId('radioButtonCircleInner')).toBeDefined();
    expect(
      result.getByTestId('radioButtonCircleInner').props.style.backgroundColor,
    ).toBe(colors.light.accent);
    expect(mockOnPress).toHaveBeenCalledTimes(0);
    fireEvent.press(result.getByTestId('radioButton'));
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
