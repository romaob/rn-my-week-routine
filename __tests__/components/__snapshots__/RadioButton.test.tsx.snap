// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton action tests should call onPress when radio button is pressed 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={6}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 3.5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 7,
            "height": 14,
            "width": 14,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="radioButtonLabel"
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`RadioButton action tests should call onPress when radio button is pressed and selected 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={7}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 3.5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "#d81e8b",
            "borderRadius": 7,
            "height": 14,
            "width": 14,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="radioButtonLabel"
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`RadioButton render tests should render the radio button 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 3.5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 7,
            "height": 14,
            "width": 14,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="radioButtonLabel"
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`RadioButton render tests should render the radio button selected 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 3.5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "#d81e8b",
            "borderRadius": 7,
            "height": 14,
            "width": 14,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="radioButtonLabel"
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`RadioButton render tests should render the radio button with label 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={3}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 3.5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 7,
            "height": 14,
            "width": 14,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="radioButtonLabel"
    >
      Test
    </Text>
  </View>
</RNGestureHandlerButton>
`;

exports[`RadioButton render tests should render the radio button with label and selected 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={4}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 3.5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "#d81e8b",
            "borderRadius": 7,
            "height": 14,
            "width": 14,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 14,
        }
      }
      testID="radioButtonLabel"
    >
      Test
    </Text>
  </View>
</RNGestureHandlerButton>
`;

exports[`RadioButton render tests should render the radio button with label and size 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={5}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  testID="radioButton"
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderColor": "#d81e8b",
          "borderRadius": 999,
          "borderWidth": 2,
          "display": "flex",
          "padding": 5,
        }
      }
      testID="radioButtonCircle"
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 10,
            "height": 20,
            "width": 20,
          }
        }
        testID="radioButtonCircleInner"
      />
    </View>
    <Text
      style={
        {
          "color": "#000000",
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 20,
        }
      }
      testID="radioButtonLabel"
    >
      Test
    </Text>
  </View>
</RNGestureHandlerButton>
`;
